pylint/reporters/ureports/nodes.py

```diff
@@ -117,17 +114,23 @@ class Section(BaseLayout):
     as a first paragraph
     """
 
-    def __init__(self, title=None, description=None, **kwargs):
+    def __init__(
+        self,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
         super().__init__(**kwargs)
         if description:
             self.insert(0, Paragraph([Text(description)]))
         if title:
             self.insert(0, Title(children=(title,)))
-        self.report_id: Optional[str] = None
+        # Used in ReportHandlerMixin make_reports
+        self.report_id: str = ""
 
 
 class EvaluationSection(Section):
-    def __init__(self, message, **kwargs):
+    def __init__(self, message: str, **kwargs: Any) -> None:
```

Comment:
Same here. I think it can be `Union[Optional[str], Union[List["Text"], Tuple[str, ...]]]`

Reply:
Apparently `` Iterable[Union["Text", str]]`` works for mypy, what do you think ? Are you sure we need the ``Optional[str]`` ?

Reply:
No, based on the idea of Marc that parameters should be as broad as possible using `Iterable` seems the right way forward! Good call!

Btw, I'm not sure if you just didn't have time to look at it yet but because the underlying comment has been resolved https://github.com/PyCQA/pylint/pull/5004#discussion_r709861703 never shows up for me on GitHub. Just want to make sure you see it :)

Reply:
Yeah I missed it as it was in a resolved conversation, thank :)